api/envoy/config/route/v3/route_components.proto

```diff
@@ -1712,6 +1712,27 @@ message RateLimit {
       Source source = 4 [(validate.rules).enum = {defined_only: true}];
     }
 
+    // The following descriptor entry is appended with a computed value.
+    // See :ref:`attributes <arch_overview_attributes>` for the set of
+    // available attributes.
+    //
+    // .. code-block:: cpp
+    //
+    //   ("<descriptor_key>", "<expression_value>")
+    message Expression {
+      // The key to use in the descriptor entry.
+      string descriptor_key = 1 [(validate.rules).string = {min_len: 1}];
+
+      oneof expr_specifier {
+        // Expression string, e.g. "connection.requested_server_name".
+        string text = 2 [(validate.rules).string = {min_len: 1}];
```

Comment:
I decided to skip parsed expr proto for now to avoid introducing a dependency from core APIs on googleapis.